modules/javafx.controls/src/test/java/test/javafx/scene/control/TreeCellTest.java

```diff
@@ -844,6 +844,33 @@ public void testEditCommitMemoryLeakAfterRemoveEditingItem() {
         assertEquals("treeItem must be gc'ed", null, itemRef.get());
     }
 
+    @Test
+    public void testStartEditOffRangeMustNotFireStartEdit() {
+        tree.setEditable(true);
+        cell.updateTreeView(tree);
+        cell.updateIndex(tree.getExpandedItemCount());
+        List<EditEvent> events = new ArrayList<>();
+        tree.addEventHandler(TreeView.editStartEvent(), events::add);
+        cell.startEdit();
+        assertFalse("sanity: off-range cell must not be editing", cell.isEditing());
+        assertEquals("cell must not fire editStart if not editing", 0, events.size());
+    }
+
+    /**
+     *  Note: this is a false green until JDK-8187474 (update control editing location) is fixed
+     */
+    @Ignore("JDK-8187474")
+    @Test
+    public void testStartEditOffRangeMustNotUpdateEditingLocation() {
+        tree.setEditable(true);
+        cell.updateTreeView(tree);
+        cell.updateIndex(tree.getExpandedItemCount());
+        cell.startEdit();
+        assertFalse("sanity: off-range cell must not be editing", cell.isEditing());
+        assertEquals("editing location", null, tree.getEditingItem());
```

Comment:
Minor: This can be replaced with `assertNull`, which would also be a bit more readable :)

Reply:
haha .. left-over from c&p (of ListCellTest ;)